src/main/java/com/esri/samples/portal/integrated_windows_authentication/IntegratedWindowsAuthenticationController.java

```diff
@@ -0,0 +1,219 @@
+          mapView.setMap(webMap);
+
+          // show progress indicator while map is drawing
+          progressIndicator.setVisible(true);
+
+          loadWebMapTextView.setText("Loaded web map from item " + selectedItem.getItemId());
+        }
+      });
+
+      // make the list view show a preview of the portal items' map area
+      resultsListView.setCellFactory(c -> new PortalItemListCell());
+
+      // hide the progress indicator when the map is finished drawing
+      mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
+        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+          progressIndicator.setVisible(false);
+        }
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Handles searching a public portal.
+   */
+  @FXML
+  private void handleSearchPublicPress() {
+    searchPortal(new Portal("http://www.arcgis.com"));
+  }
+
+  /**
+   * Handles searching a secure portal.
+   */
+  @FXML
+  private void handleSearchSecurePress() {
+    // check for a url in the URL field
+    if (!portalUrlTextField.getText().isEmpty()) {
+      // search an instance of the IWA-secured portal, the user may be challenged for access
+      searchPortal(new Portal(portalUrlTextField.getText(), true));
+    } else {
+      new Alert(Alert.AlertType.ERROR, "Portal URL is empty. Please enter a portal URL.").show();
+    }
+  }
+
+  /**
+   * Search the given portal for its portal items and display them in a list view.
+   *
+   * @param portal to search
+   */
+  private void searchPortal(Portal portal) {
+
+    // check if the portal is null
+    if (portal == null) {
+      new Alert(Alert.AlertType.ERROR, "No portal provided").show();
+      return;
+    }
+
+    // clear any existing items in the list view
+    resultsListView.getItems().clear();
+
+    // clear the information about the previously loaded map
+    loadWebMapTextView.setText("");
+
+    // show portal load state
+    progressIndicator.setVisible(true);
+    loadStateTextView.setText("Searching for web map items on the portal at " + portal.getUri());
+
+    // load the portal items
+    portal.loadAsync();
+    portal.addDoneLoadingListener(() -> {
+      if (portal.getLoadStatus() == LoadStatus.LOADED) {
+        try {
+          // update load state in UI with the portal URI
+          loadStateTextView.setText("Connected to the portal on " + new URI(portal.getUri()).getHost());
+        } catch (URISyntaxException e) {
+          new Alert(Alert.AlertType.ERROR, "Error getting URI from portal: " + e.getMessage()).show();
+        }
+
+        // report the user name used for this connection
+        if (portal.getUser() != null) {
+          loadStateTextView.setText("Connected as: " + portal.getUser().getUsername());
+        } else {
+          // if connecting to an unsecured portal, no user name is needed
```

Comment:
yup, this is a better comment. I think we could be even more specific with the following suggestion, up to you though if you want to add that in.
"if connecting to an unsecured portal, no user credentials are needed to authenticate access"

Reply:
More specific comments are always better 👍